keps/sig-auth/2579-pod-isolation-policy/README.md

```diff
@@ -0,0 +1,951 @@
+    - Enforcing: policy violations cause the pod to be rejected
+    - Audit: policy violations trigger an audit annotation, but are otherwise allowed
+    - Warning: policy violations trigger a user-facing warning, but are otherwise allowed
+- An optional per-mode version label can be used to pin the policy to the version that shipped
+  with a given Kubernetes minor version (e.g. `v1.18`)
+- Dry-run of namespace updates is supported to test enforcing policy changes against existing pods.
+- Policy exemptions can be statically configured based on (requesting) user, RuntimeClass, or
+  namespace. A request meeting exemption criteria is ignored by the admission plugin.
+
+[Pod Security Standards]: https://kubernetes.io/docs/concepts/security/pod-security-standards/
+
+## Motivation
+
+Pod Security Policy is deprecated as of Kubernetes v1.21. There were numerous problems with
+PSP that lead to the decision to deprecate it, rather than promote it to GA, including:
+
+1. Policy authorization model - Policies are bound to the requesting user OR the pod’s service
+   account
+     - Granting permission to the user is intuitive, but breaks controllers
+     - Dual model weakens security
+2. Rollout challenges - PSP fails closed in the absence of policy
+     - The feature can never be enabled by default
+     - Need 100% coverage before rolling out (and no dry-run / audit mode)
+     - Leads to insufficient test coverage
+3. Inconsistent & unbounded API - Partly an aesthetic issue, but highlights lack of flexibility
+     - API has grown organically and has many internal inconsistencies (usability challenge)
+     - Unclear how to decide what should be part of PSP (e.g. fine-grained volume restrictions)
+     - Doesn’t compose well
+     - Mutation priority can be unexpected
+
+However, we still feel that Kubernetes should include a mechanism to prevent privilege escalation
+through the create-pod permission.
+
+### Goals
+
+Replace PodSecurityPolicy without compromising the ability for Kubernetes to limit privilege
+escalation out of the box. Specifically, create/update pod permission should not be equivalent
+to root-on-node (or cluster).
+
+#### Requirements
+
+1. Validating only (i.e. no changing pods to make them comply with policy)
+2. Safe to enable in new AND upgraded clusters
+    - Dryrun policy changes and/or Audit-only mode
+3. Built-in in-tree controller
+4. Capable of supporting Windows in the future, if not in the initial release
+    - Don’t automatically break windows pods
+5. Must be responsive to Pod API evolution across versions
+6. (fuzzy) Easy to use, don’t need to be a kubernetes/security/linux expert to meet the basic objective
+
+Nice to have:
+
+1. Exceptions or policy bindings by requesting user
+2. Extensible: should work with custom policy implementations without whole-sale replacement
+3. Windows support in the initial release
+4. Enabled by default (not in alpha)
```

Comment:
```suggestion
4. Can be enabled by default in beta/GA phase
```

Reply:
Why "can be"? I guess there's 2 different ways to interpret this item: the admission controller defaults to enforcing baseline on unlabeled namespaces, or the admission controller is in the default set of admission controllers.